packages/plugin-core/src/KeybindingUtils.ts

```diff
@@ -1,206 +1,330 @@
+  }) {
+    const { knownConflicts } = opts;
+    const installStatus = KeybindingUtils.getInstallStatus();
+    const installed = installStatus
+      .filter((status) => status.installed)
+      .map((status) => status.id);
+
+    const conflicts = knownConflicts.filter((conflict) => {
+      const isInstalled = installed.includes(conflict.extensionId);
+      const osType = os.type();
+      const conflictOSType = conflict.os || ["Darwin", "Linux", "Windows_NT"];
+      const matchesOS = isOSType(osType) && conflictOSType.includes(osType);
+      return isInstalled && matchesOS;
+    });
 
-  static checkAndApplyVimKeybindingOverrideIfExists(): RespV3<{
-    keybindingConfigPath: string;
-    newKeybindings?: any;
-  }> {
-    // check where the keyboard shortcut is configured
+    // for each of the found conflicts, see if the user has them remapped / disabled  in keybinding.json
     const { keybindingConfigPath } = this.getKeybindingConfigPath();
+    const userKeybindingConfigExists = fs.existsSync(keybindingConfigPath);
 
-    const resp = this.getKeybindingConfig({ createIfMissing: true });
-    if (resp.error) {
-      return {
-        error: DendronError.createFromStatus({
-          status: ERROR_STATUS.INVALID_STATE,
-          message: "no keybindings found",
-        }),
-      };
+    // all conflicts are valid
+    if (!userKeybindingConfigExists) {
+      return conflicts;
     }
-    const keybindings = resp.data;
-
-    // check if override is already there
-    const alreadyHasOverride =
-      keybindings.filter((entry: any) => {
-        if (!_.isUndefined(entry.command)) {
-          return entry.command === "-extension.vim_navigateCtrlL";
-        } else {
-          return false;
-        }
-      }).length > 0;
 
-    if (alreadyHasOverride) {
-      return { data: { keybindingConfigPath } };
+    const userKeybindingConfig = readJSONWithCommentsSync(
+      keybindingConfigPath
+    ) as CommentJSONArray<Keybindings>;
+
+    const alreadyResolved: KeybindingConflict[] = [];
+
+    userKeybindingConfig.forEach((keybinding) => {
+      // find remap / disable of conflicting keybindings
+      let command = keybinding.command;
+      if (command.startsWith("-")) {
+        command = command.substring(1);
+      }
+
+      const resolvedConflict = conflicts.find(
+        (conflict) => conflict.commandId === command
+      );
+      if (resolvedConflict) {
+        alreadyResolved.push(resolvedConflict);
+      }
+
+      // find remap / disable of dendron keybindings
+      command = keybinding.command;
+      if (command.startsWith("-")) {
+        command = command.substring(1);
+      }
+      const resolvedDendronConflict = conflicts.find(
+        (conflict) => conflict.conflictsWith === command
+      );
+
+      if (resolvedDendronConflict) {
+        alreadyResolved.push(resolvedDendronConflict);
+      }
+    });
+
+    return _.differenceBy(conflicts, alreadyResolved);
```

Comment:
this seems very expensive to run on each loop. why can't we just add all conflicts to a set and report if the set is non empty?